catroid/src/main/java/org/catrobat/catroid/ui/MainMenuActivity.java

```diff
@@ -116,15 +117,7 @@
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
-		if (getIntent().getExtras() != null) {
-			try {
-				String link = getIntent().getExtras().get("link").toString();
-				startActivity(new Intent(Intent.ACTION_VIEW, Uri.parse(link)));
-			} catch (NullPointerException e) {
-				Log.e(TAG, "Url not found with notification", e);
-			}
-		}
-
+		getUrlFromCloudMessage();
```

Comment:
`getX` without returning something isn't normally expected. I first thought about `openUrlFromCloudMessage` but that's not a good name. What do you think about `checkCloudMessage` or `handleOpenedCloudMessage`? Maybe you can come up with something better.